azext_iot/central/providers/device_provider.py

```diff
@@ -191,11 +194,14 @@ def get_device_registration_info(
             dps_state = dps_global_service.get_registration_state(
                 id_scope=id_scope, key=key, device_id=device_id
             )
-        dps_state = self.dps_populate_essential_info(dps_state, device.device_status)
+        dps_state = self.dps_populate_essential_info(
+            dps_state, device.get_registration_info().get("device_status")
```

Comment:
I think this can be reduced to `dps_state = self.dps_populate_essential_info(dps_state, device.device_status)`